### DISCUSSION TITLE
Icon Order

### DISCUSSION THREAD

**Discussion by madprops** [madprops/grasshopper#159] [General]
Big changes regarding icon positions.

Say you wanted an icon to be more to the left, or more to the right, it's now possible

Just re-order them in the list:

![iconorder](https://github.com/user-attachments/assets/13ea52c3-27ff-461a-870d-3134f8fc6808)

Top of the list means more to the left.

**Comment by N3C2L** [comment]
Very good! I love these icon enhancements! I use the icons daily! 👍 

What do you think of the idea of associating custom commands with icons? These custom icons could be linked to a custom command that the user can create somewhere in the settings. After creating this icon + command combination in the settings, a user could then right-click a tab and add this icon that they previously associated with a command. This custom icon would then be visible on the tab along with the other icons. And could also be reordered with your cool new icon order enhancements! I'd love this 👍

**Comment by madprops** [comment]
I see what you mean. But not sure how that would be better than executing a command on an item through other ways. What would the icons represent?

Would be easier to define commands for the custom icons.

For instance `🤖` == `toggle red when clicked`

Only drawback is that all custom icons share the same order

But in general sounds a bit messy to do something like that

I'll try giving custom icons specific commands, sounds fun

**Reply by N3C2L** [reply]
> I'll try giving custom icons specific commands, sounds fun

That's great to hear! Thanks! Looking forward to test this when implemented 👍

**Comment by madprops** [comment]
Added icons

![icon234](https://github.com/user-attachments/assets/2da738dd-12bf-4341-a980-db10c00986a1)
